### PR TITLE
Added support to stream App logs for Windows container when containerLogs is called

### PR DESCRIPTION
**Pull request by pagariyaalok**
Added support to stream App logs for Windows container when containerLogs is called
Few clean up and performance optimization.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/projectkudu/kudu?pullRequest=3176) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/projectkudu/kudu?pullRequest=3176) before we can accept your contribution.<br/><br/>:x: pagariyaalok [sign now](https://cla.dotnetfoundation.org/projectkudu/kudu?pullRequest=3176)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/projectkudu/kudu?pullRequest=3176) it.</sub>

**Comment by pagariyaalok** [comment]
@suwatch

**Comment by pagariyaalok** [comment]
@vinisoto

**Comment by pagariyaalok** [comment]
@joaquinv is added to the review. #Closed

**Comment by pagariyaalok** [comment]
@suwatch is added to the review. #Closed

**Comment by pagariyaalok** [comment]
@vinisoto is added to the review. #Closed

**Comment by pagariyaalok** [comment]
@jvano is added to the review. #Closed

**Comment by jvano** [comment]
:shipit:

File: Kudu.Services/Diagnostics/DiagnosticsController.cs [hunk 1]
```diff
@@ -79,53 +85,53 @@ public HttpResponseMessage GetRecentLogs(int top = 100)
...
+            using (_tracer.Step("DiagnosticsController.GetContainerLogsZip"))
             {
-                // Also search for "today's" files in sub folders. Windows containers archives log files
-                // when they reach a certain size.
-                var currentDockerLogFilenames = GetCurrentDockerLogFilenames(SearchOption.AllDirectories);
+                var currentContainerLogFilenames = GetCurrentContainerLogFilenames(SearchOption.TopDirectoryOnly);
 
                 HttpResponseMessage response = Request.CreateResponse();
-                response.Content = ZipStreamContent.Create(String.Format("dockerlogs-{0:MM-dd-HH-mm-ss}.zip", DateTime.UtcNow), _tracer, zip =>
+                response.Content = ZipStreamContent.Create(String.Format("containerlogs-{0:MM-dd-HH-mm-ss}.zip", DateTime.UtcNow), _tracer, zip =>
```
Comment (vinisoto) [inline Kudu.Services/Diagnostics/DiagnosticsController.cs:132]: If we change the name of the .zip file we may break customers (for example: if they have automation to download the logs periodically)

**Comment by pagariyaalok** [comment]
@JennyLawrance is added to the review. #Closed

**Comment by pagariyaalok** [comment]
@sanchitmehta is added to the review. #Closed

File: Kudu.Services/Diagnostics/DiagnosticsController.cs (hunk 1 above)
Comment (pagariyaalok) [reply Kudu.Services/Diagnostics/DiagnosticsController.cs:132]: yes I thought of it, but we dont have any better option..We dont want to keep it as docker logs for sure. Open for ideas...

**Pull request merged**